client/src/components/SentimentAnalysis/SentimentAnalysis.js

```diff
@@ -72,16 +94,64 @@ const SentimentAnalysis = props => {
    * - If not, show empty textarea so user can filled in themselves
    */
   const displayTextareaValue = label => {
-    if (locationState === undefined) {
+    if (id === undefined) {
       return '';
     }
-    return label === 'headline'
-      ? locationState.ad.headline
-      : locationState.ad.content;
+    return label === 'headline' ? headline.text : content.text;
+  };
+
+  /**
+   * Generates a PNG screenshot of the current result page and
+   * automatically saves it to user's local machine
+   * Note: download as PNG because of its flexibility in dimensions
+   * @returns {Void} doesn't return anything
+   */
+  const downloadScreenshot = () => {
+    const input = document.getElementById('analysis-container');
+    html2canvas(input).then(canvas => {
+      const imgData = canvas.toDataURL('image/png');
+      saveAs(imgData, 'tardigrade-analysis.png');
+    });
+  };
+
+  useEffect(() => {
+    if (id === undefined) {
+      setShareLink(undefined);
+      return;
+    }
+    database
+      .collection('ads')
+      .doc(id)
+      .get()
+      .then(doc => updatePage(doc.data()))
+      .catch(error => console.log('Error getting documents: ', error));
+  }, [id, headline, content, shareLink]);
```

Comment:
Why are you accessing the DB? Not entirely clear to me why you'd do this.

Reply:
To get the Ad details given the ID in route's param